src/plugins/xfer/xfer-dcc.c

```diff
@@ -368,9 +368,14 @@ xfer_dcc_recv_file_child (struct t_xfer *xfer)
         ready = poll (&poll_fd, 1, -1);
         if (ready <= 0)
         {
-            xfer_network_write_pipe (xfer, XFER_STATUS_FAILED,
-                                     XFER_ERROR_RECV_BLOCK);
-            return;
+            if ((errno == EINTR) || (errno == EAGAIN))
```

Comment:
Seems like xfer_network_write_pipe() would be right place to check the result of write(). You ought to be able to immediately retry if you get EINTR (see glibc's TEMP_FAILURE_RETRY)